src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -32,7 +32,7 @@ namespace NuGet.PackageManagement.VisualStudio
     /// <remarks>Each concrete implementation is responsible of initializing <see cref="InstalledPackages"/> and <see cref="TransitivePackages"/> collections</remarks>
     /// <typeparam name="T">A collection type for Installed and Transtive packages</typeparam>
     /// <typeparam name="U">Type of the collection elements for Installed and Transitive packages</typeparam>
-    public abstract class PackageReferenceProject<T, U> : BuildIntegratedNuGetProject, IPackageReferenceProject where T : ICollection<U>
+    public abstract class PackageReferenceProject<T, U> : BuildIntegratedNuGetProject, IPackageReferenceProject where T : ICollection<U>, new()
```

Comment:
Please add a test for GetInstalledPackages() in which two tasks run that method at the same time, and results should be the same, but with different objects.

Reply:
I'll add more tests after getting initial approval, if I have to change code then tests change too.

Reply:
Not sure, if we can test that way.